mqtt_client.c

```diff
@@ -39,21 +39,21 @@ static int mqtt_message_receive(esp_mqtt_client_handle_t client, int read_poll_t
 static void esp_mqtt_client_dispatch_transport_error(esp_mqtt_client_handle_t client);
 static esp_err_t send_disconnect_msg(esp_mqtt_client_handle_t client);
 
-static int esp_mqtt_handle_transport_read_error(int err, esp_mqtt_client_handle_t client)
+static void esp_mqtt_handle_transport_read_error(int err, esp_mqtt_client_handle_t client)
 {
-    if (err == ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN) {
-        ESP_LOGD(TAG, "%s: transport_read(): EOF", __func__);
-        return 0;
```

Comment:
Hi @egnor , i think regard `ERR_TCP_TRANSPORT_CONNECTION_TIMEOUT` as error is not suitable. Do you meet any case that need regard time-out as error? I think when err is `ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN`, we need `return -1` here. Could you please only change code with `return -1` here to test the case you reported?

Reply:
This function is now `void` and no longer returns anything at all. (Are you sure you're looking at the changed version of the code? This annotation is on the old version.)

Whether `ESP_TCP_TRANSPORT_CONNECTION_TIMEOUT` is an error or not depends on context, and that case is handled in `mqtt_message_receive()` below, where it's treated as an error mid-message but not an error for the first byte.

Can you clarify your request?

Reply:
@egnor Yes, i mean that on the old version, which not include your changes, only modify the one line that when `err` is `ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN`, we return `-1` instead of `0`.

When we handle the mid-message and catch the `ESP_TCP_TRANSPORT_CONNECTION_TIMEOUT `, we still can return `0` and not treate it as error. And then we will continue call `mqtt_message_receive` to read the data for mid-message. As far as I understand, the first byte timeout and the subsequent byte timeout can actually be considered not errors. So I want to know whether you meet the problem, if you don’t handle mid-message time-out as error?

Reply:
Ah, I see what you mean.

It's important that `CLOSED_BY_FIN` and other such "hard" errors are always treated as errors, whether they happen at the start of the message or mid-message.

As for whether `CONNECTION_TIMEOUT` is an error, even if midmessage-- well that's something of a policy matter, but if you can ensure that it is handled gracefully, and that keepalive failure is still handled correctly, then I suppose it could work if there's something like a very slow network connection.